Fixes:

Samba client is unable to write to a file it just created.

Samba drops privileges and tries to re-open a file descriptor for writing via
/proc filesystem. The file was originally opened as root, but re-open is done
as common user. The results is EACCES on Solaris.

The workaround is to avoid re-opening a file using /proc filesystem
on Solaris. Instead fall back to fd_open_atomic().

--- samba-4.20.5/source3/smbd/open.c
+++ samba-4.20.5/source3/smbd/open.c
@@ -1171,6 +1171,7 @@ {
 	NTSTATUS status;
 	int old_fd;
 
+#if 0
 	if (fsp->fsp_flags.have_proc_fds &&
 	    ((old_fd = fsp_get_pathref_fd(fsp)) != -1)) {
 
@@ -1211,6 +1212,7 @@ }
 		fsp_set_fd(fsp, new_fd);
 		return NT_STATUS_OK;
 	}
+#endif
 
 	/*
 	 * Close the existing pathref fd and set the fsp flag
